Common/ArmEmitter.cpp

```diff
@@ -268,7 +268,7 @@ bool ARMXEmitter::TryANDI2R(ARMReg rd, ARMReg rs, u32 val)
 		}
 		return true;
 	} else {
-#ifdef HAVE_ARMV7
+#if PPSSPP_ARCH(ARMV7)
```

Comment:
Yes, it should be easy to make it a runtime check by adding an armv7 field in the cpu_info struct (which could be populated by looking at the current arch we compile for).
Runtime CPU detection on ARM is tricky though, so we'll need to be careful.